viewing_party/main.py

```diff
@@ -0,0 +1,142 @@
+    movies_dict = {}
+    if title and genre and rating:
+        movies_dict["title"] = title
+        movies_dict["genre"] = genre
+        movies_dict["rating"] = rating
+        return movies_dict
+    elif not title or not genre or not rating:
+        return None
+
+def add_to_watched(user_data, movie):
+    """Add a movie to user's 'WATCHED' list."""
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    """Add a movie to a user's 'WATCHLIST'."""
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    """Remove watched movies from the user's 'WATCHLIST' 
+        and add them to the 'WATCHED' list. 
+        """
+    for movie_in_list in user_data["watchlist"]:
+    # my_index allows us to access elements in the list we are working with
+        my_index = len(user_data["watchlist"]) 
+        if title in user_data["watchlist"][my_index-1]["title"]:
+            movie_data = user_data["watchlist"][my_index-1]
+            user_data["watchlist"].remove(movie_data)
+            user_data["watched"].append(movie_data)
+            return user_data
+        elif title not in user_data["watchlist"][my_index-1]["title"]:
+            return user_data
+
+def get_watched_avg_rating(user_data):
+    """Calculate average rating of movies in user's 'WATCHED' list."""
+    hold_ratings = []
+    for movie_in_watched in user_data["watched"]:
+        rating = movie_in_watched["rating"]
+        hold_ratings.append(rating)
+    try: 
+        divisor = len(user_data["watched"])
+        avg_rating = sum(hold_ratings) / (divisor)
+    except ZeroDivisionError as err:
+        return 0.0
+    return avg_rating 
+
+def get_most_watched_genre(user_data):
+    """Determine and return the genre the user watches most frequently."""
+    hold_genres = {}
+    if user_data["watched"] == []:
+            return None        
+    for movies_in_watched in user_data["watched"]:
+        genre = movies_in_watched["genre"] 
+        if genre not in hold_genres:
+            hold_genres[genre] = 1
+        else:
+            hold_genres[genre] += 1
+    most_frequent = max(hold_genres)
+    return most_frequent 
+
+def get_unique_watched(user_data):
+    """Determine and return a list of movies that the user has watched
+        but that the user's friends have not watched.
+        """
+    movies_user_watched = []
+    movies_friends_watched = []
+    unique_titles = []
+    for movie_info in user_data["watched"]: 
+        movies_user_watched.append(movie_info) 
+    for friend in user_data["friends"]:
+        for friend_movie_info in friend["watched"]:
+            movies_friends_watched.append(friend_movie_info)
+    for entry in movies_user_watched:
+        if entry not in movies_friends_watched:
+            unique_titles.append(entry)
+    return unique_titles
+
+def get_friends_unique_watched(user_data):
```

Comment:
👍 , do you think you could re-use a previous function or create a helper function.  I'm seeing some duplicated code.